angular/core/components/change_volume_form/change_volume_form.coffee

```diff
@@ -0,0 +1,42 @@
+angular.module('loomioApp').factory 'ChangeVolumeForm', ->
+  templateUrl: 'generated/components/change_volume_form/change_volume_form.html'
+  controller: ($scope, model, setDefault, FormService, CurrentUser, FlashService) ->
+    $scope.model = model.clone()
+    $scope.volumeLevels = ["loud", "normal", "quiet"]
+    $scope.buh = {}
+    $scope.buh.volume = model.volume()
+    $scope.setDefault = setDefault if setDefault
+
+    $scope.translateKey =
+      if setDefault
+        "change_volume_form.set_membership_default"
+      else
+        "change_volume_form.#{$scope.model.constructor.singular}"
+
+    $scope.title =
+      if setDefault
+        ''
+      else
+        model.title or model.groupName()
+
+
+    $scope.submit = ->
+      $scope.isDisabled = true
```

Comment:
We'll want to move this into the FormService; I think something like this will work:

```
$scope.submit = FormService.submit $scope, $scope.model,
  submitFn: (model) ->
    model.saveVolume($scope.buh.volume, $scope.applyToAll, $scope.setDefault
  flashSuccess: $scope.someFunctionToGetTheFlashSuccess
```